### PR TITLE
Fix #3586 Improve date/time/filter in feature grid

### LINKED ISSUE #3586: Improve Date/Time/DateTime ColumnFilter in FeatureGrid
# Description
Now column filters of the feature grid have a minimal support for date/time and datetime values. 
 - Do not allow > < = <>(!=) filters, filter only on equal date
 - The data inserted by the user is inteded in his datetime, not UTC.

We should improve it to align with functionalities of number filters and so, support better data text insert.  

### Acceptance Criteria
 - User should be able to insert manually date time, intended as UTC
 - Format of writing should be configurable, depending on user's locale and shown as hint in the placeholder
 - User can insert an operator (> < = <>) before the date to change the filter accordingly (intended as attribute > date_written)
 - The user can use the picker to insert the date/time/datetime
 - User can reset the filter clearing the text
 - TBD - Default insert date format for date/time/datetime

### Tasks
- [ ] Improve text datetime insert functionalities (now user can insert data in his browser's language format). 2PT
- [ ] Allow opearators >, <, =, !=(<>) operators ( requires to improve format and insert ) 1PT
- [ ] Fix issues with data insert
  - with datetime attribute, if you insert only the date, the time will be the current hour, instead it should be 00:00:00 1PT
- [ ] Fix data insert to use UTC (now current browser's time, e.g. GMT+1, then converted in UTC). 1PT

### PR DESCRIPTION
**Pull request by kasongoyo**
Feature grid date filter was not allowing less/equal/greater operators.
This commit add the ability for use to type in operator to be used within
date filter in feature grid

## Description
A few sentences describing the overall goals of the pull request' s commits.

## Issues

 - #3586

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [ ] Yes, and I documented them in migration notes
 - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: web/client/components/data/featuregrid/filterRenderers/baseDateTimeFilter/BaseDateTimeFilter.js [hunk 1]
```diff
@@ -1,31 +1,34 @@
...
 })(DateTimePicker);
 
+
 class DateFilter extends AttributeFilter {
     static propTypes = {
         type: PropTypes.string,
         disabled: PropTypes.boolean,
-        onChange: PropTypes.func
+        onChange: PropTypes.func,
+        intl: intlShape.isRequired
```
Comment (offtherailz) [inline web/client/components/data/featuregrid/filterRenderers/baseDateTimeFilter/BaseDateTimeFilter.js]: Please find out a way to wrap i18n in a separate  HOC outside, to allow reuse of DateTiimeFilter (and DateTimePicker) and avoid to redefine everything.

File: web/client/components/data/featuregrid/filterRenderers/__tests__/BaseDateTimeFilter-test.jsx [hunk 2]
```diff
@@ -30,37 +30,37 @@ describe('Test for BaseDateTimeFilter component', () => {
...
         expect(el).toExist();
 
     });
-    it('Test BaseDateTimeFilter onChange', () => {
-        const actions = {
-            onChange: () => {}
-        };
-        const spyonChange = expect.spyOn(actions, 'onChange');
-        const cmp = ReactDOM.render(<BaseDateTimeFilter type="date" onChange={actions.onChange} />, document.getElementById("container"));
+    // it.skip('Test BaseDateTimeFilter onChange', () => {
```
Comment (offtherailz) [inline web/client/components/data/featuregrid/filterRenderers/__tests__/BaseDateTimeFilter-test.jsx]: Do not comment code. If this test is not valid anymore, remove it.

File: web/client/components/misc/datetimepicker/index.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+// module.exports = require('./DateTimePicker');
```
Comment (offtherailz) [inline web/client/components/misc/datetimepicker/index.js]: remove commented code
```suggestion
```

File: web/client/components/misc/datetimepicker/DateTimePicker.js [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+    return value.toDate();
+};
+
+const formats = {
+    base: 'lll',
+    date: 'L',
+    time: 'LT'
+};
+
+class DateTimePicker extends Component {
```
Comment (offtherailz) [inline web/client/components/misc/datetimepicker/DateTimePicker.js]: This DateTimePicker supports also the operator. Please document this a little bit.

**Comment by kasongoyo** [comment]
- [ ] onChange event management of the operator: https://coveralls.io/builds/23946165/source?filename=web%2Fclient%2Fcomponents%2Fmisc%2Fdatetimepicker%2FDateTimePicker.js#L29

*From the link, it's clear that the test required is for calendar onChange or time onChange event something difficult to simulate or perhaps it will need more time but both of the event have been covered with tests in their respective component.*

- [ ] I see coverage is influenced also by key events management in hours selector https://coveralls.io/builds/23946165/source?filename=web%2Fclient%2Fcomponents%2Fmisc%2Fdatetimepicker%2FHours.js#L48 

*I will add more tests on this*

**Pull request merged**